godot-core/src/builtin/color.rs

```diff
@@ -4,30 +4,511 @@
+        Self::from_rgba(r, g, b, 1.0)
+    }
+
+    /// Constructs a new `Color` with the given components as bytes. 0 is mapped to 0.0, 255 is
+    /// mapped to 1.0. Equivalent of the global constructor function `Color8` in GDScript.
+    pub fn from_rgba8(r: u8, g: u8, b: u8, a: u8) -> Self {
+        Self::from_rgba(from_u8(r), from_u8(g), from_u8(b), from_u8(a))
+    }
+
+    /// Constructs a new `Color` with the given components as bytes. 0 is mapped to 0.0, 65535
+    /// (`0xffff`) is mapped to 1.0.
+    pub fn from_rgba16(r: u16, g: u16, b: u16, a: u16) -> Self {
+        Self::from_rgba(from_u16(r), from_u16(g), from_u16(b), from_u16(a))
+    }
+
+    /// Constructs a new `Color` from a 32-bits value with the given channel `order`. Passing
+    /// `ColorChannelOrder::Rgba` is the equivalent of `Color.hex` in GDScript.
+    pub fn from_u32_rgba(u: u32, order: ColorChannelOrder) -> Self {
+        let [r, g, b, a] = order.unpack(u.to_be_bytes());
+        Color::from_rgba8(r, g, b, a)
+    }
+
+    /// Constructs a new `Color` from a 64-bits value with the given channel `order`. Passing
+    /// `ColorChannelOrder::Rgba` is the equivalent of `Color.hex64` in GDScript.
+    pub fn from_u64_rgba(u: u64, order: ColorChannelOrder) -> Self {
+        let [r, g, b, a] = order.unpack(to_be_words(u));
+        Color::from_rgba16(r, g, b, a)
+    }
+
+    /// Constructs a `Color` from an HTML color code string. Valid values for the string are:
+    ///
+    /// - `#RRGGBBAA` and `RRGGBBAA` where each of `RR`, `GG`, `BB` and `AA` stands for two hex
+    ///   digits (case insensitive).
+    /// - `#RRGGBB` and `RRGGBB`. Equivalent to `#RRGGBBff`.
+    /// - `#RGBA` and `RGBA` where each of `R`, `G`, `B` and `A` stands for a single hex digit.
+    ///   Equivalent to `#RRGGBBAA`, i.e. each digit is repeated twice.
+    /// - `#RGB` and `RGB`. Equivalent to `#RRGGBBff`.
+    ///
+    /// Returns `None` if the format is invalid.
+    pub fn from_html<S: Into<GodotString>>(html: S) -> Option<Self> {
+        let html = html.into();
+        InnerColor::html_is_valid(html.clone()).then(|| InnerColor::html(html))
+    }
+
+    /// Constructs a `Color` from a string, which can be either:
+    ///
+    /// - An HTML color code as accepted by [`Color::from_html`].
+    /// - The name of a built-in color constant, such as `BLUE` or `lawn-green`. Matching is case
+    ///   insensitive and hyphens can be used interchangeably with underscores. See the [list of
+    ///   color constants][color_constants] in the Godot API documentation, or the visual [cheat
+    ///   sheet][cheat_sheet] for the full list.
+    ///
+    /// Returns `None` if the string is neither a valid HTML color code nor an existing color name.
+    ///
+    /// [color_constants]: https://docs.godotengine.org/en/latest/classes/class_color.html#constants
+    /// [cheat_sheet]: https://raw.githubusercontent.com/godotengine/godot-docs/master/img/color_constants.png
```

Comment:
Linking this is great, thanks! Is it still up-to-date for Godot 4?

Reply:
The links go straight to the `latest` docs, so if anything, it'll be _too_ up-to-date :)